azure-spring-boot/src/main/java/com/microsoft/azure/spring/cloundfoundry/environment/VcapProcessor.java

```diff
@@ -76,27 +67,33 @@ public VcapResult parse(String vcapServices) {
         log("VcapParser.parse:  vcapServices = " + vcapServices);
         if (vcapServices != null) {
             try {
-                final JSONObject json = new JSONObject(vcapServices);
-                final JSONArray names = json.names();
+                final ObjectMapper objectMapper = new ObjectMapper();
+                objectMapper.configure(DeserializationFeature.UNWRAP_ROOT_VALUE, false);
+
+                final Map<String, List<VcapServiceConfig>> servicesMap = objectMapper.readValue(vcapServices,
+                        new TypeReference<Map<String, List<VcapServiceConfig>>>() { });
+                final Set<Map.Entry<String, List<VcapServiceConfig>>> services = servicesMap.entrySet();
+
+                if (services != null) {
+                    final Iterator<Map.Entry<String, List<VcapServiceConfig>>> serviceIterator = services.iterator();
+
+                    while (serviceIterator.hasNext()) {
+                        final Map.Entry<String, List<VcapServiceConfig>> serviceEntry = serviceIterator.next();
+                        final String name = serviceEntry.getKey();
 
-                if (names != null) {
-                    for (int i = 0; i < names.length(); i++) {
-                        final String name = (String) names.get(i);
                         if (name.startsWith(AZURE) || USER_PROVIDED.equals(name)) {
-                            final JSONArray azureService = json.getJSONArray(name);
-                            final int numElements = azureService.length();
-                            for (int index = 0; index < numElements; index++) {
-                                final VcapPojo pojo = parseService(name, azureService,
-                                        vcapServices, index);
-                                
+                            final List<VcapServiceConfig> azureServices = serviceEntry.getValue();
```

Comment:
I am not sure here we can change to this ?
```java
results.addAll(azureServices.stream().map(s -> parseService(name, s, vcapServices).filter(p -> p != null))
```

Reply:
Updated with lambda expression.